New theme

devem-software opened the pull request:
Add new theme
Remove fontawesome icons
Add material design icons

devem-software committed:
New theme

devem-software committed:
Fix Settings.mjs

Sheepux commented on src/web/mjs/engine/Settings.mjs:
```diff
@@ -57,9 +57,10 @@ export default class Settings extends EventTarget {
+                { value: 'frontend@modern', name: 'Modern <devemloop>' }
             ],
-            value: 'frontend@classic-light'
+            value: 'frontend@modern'
```
I think we should stay on classic-light by default -for a while- so that more "early adopters" can test it thouroughly.

27136773 commented on src/web/css/materialdesignicons/.editorconfig:
```diff
@@ -0,0 +1,8 @@
+root = true
```
Do not check in user configuration files => delete this file or add entry to `.gitignore`

27136773 commented on src/web/index.html:
```diff
@@ -14,161 +18,233 @@
     <script src="./js/ass.min.js" defer></script>
     <script src="./js/sql.min.js" defer></script>
     <script src="./js/exif-js.min.js" defer></script>
+    <link rel="stylesheet" href="./css/materialdesignicons/mdi.css" />
```
Use this directly in the theme of the new frontend, the index should not apply any theme based styles

27136773 reviewed (commented):
Briefly checked all the files, i think the `index.html` should only contain changed that are technically necessary to load the themes. It should not add any style or theme itself.

devem-software committed:
Update Modern Theme

Fix some bugs
Refactor components code
Deleted unnecesary files

27136773 commented on src/web/index.html:
```diff
@@ -1,250 +1,175 @@
     <script src="./js/oauth-1.0a.min.js"></script>
     <script src="./js/jszip.min.js" defer></script>
-    <script src="./js/pdfkit.standalone.js" defer></script>
+    <script src="./js/jspdf.min.js" defer></script>
```
`jspdf` was replaced by `pdfkit`, so it should not be reverted

27136773 commented on src/web/index.html:
```diff
@@ -1,250 +1,175 @@
     <script src="./js/hls.light.min.js" defer></script>
     <script src="./js/ass.min.js" defer></script>
     <script src="./js/sql.min.js" defer></script>
-    <script src="./js/exif-js.min.js" defer></script>
```
`exif` is required by one of the connectors to extract information from an image required for scraping and therefore should not be removed

27136773 commented on src/web/index.html:
```diff
@@ -148,6 +148,7 @@
         }
 
         #default_menubar {
+            width: 100%;
```
`menubar` is a block element (div) and therefore setting a width of 100% is not explicitly required

devem-software replied:
the setting is done because the body has the flex property in its css

27136773 commented on src/web/index.html:
```diff
@@ -111,11 +110,12 @@
     </script>
     <style>
         html, body {
+            display:flex;
```
Is there any benefit in using flex?

devem-software replied:
All the modern style uses flexbox, for this reason it is required that the body and the html have this tag for correct operation, the flex display does not affect the other styles

27136773 reviewed (commented):
Can be merged after fixing the included scripts in `index.html`

devem-software committed:
Fix some styles

Fix some styles & update vcp

27136773 commented on src/web/mjs/connectors/VerComicsPorno.mjs:
```diff
@@ -1,55 +1,66 @@
-import Connector from '../engine/Connector.mjs';
```
This file is not related to this PR and should be updated in a separate PR

27136773 commented on src/web/mjs/connectors/TvYManga.mjs:
```diff
@@ -33,6 +33,7 @@ export default class TvYManga extends Connector {
     async _getChapters(manga) {
```
This file is not related to this PR and should be updated in a separate PR

devem-software committed:
Some fixes

Fixed styles in 
pageComponent/style.css
chaptersComponent/style.css
mangasComponent/style.css

devem-software committed:
Merge remote-tracking branch 'origin/modern-theme' into modern-theme

devem-software committed:
Change Icon in pagesComponent

Uniform Style icons in actions buttons for pagesComponent

devem-software committed:
Revert "Merge remote-tracking branch 'origin/modern-theme' into modern-theme"

This reverts commit 6056418dd717b33b872096397d84229521ad0ad7, reversing
changes made to d4398a3dcd5b314aac6ea596e670cad56b91ddb1.

devem-software committed:
Fix index

devem-software committed:
Update frontend@modern

### Added
- The i18n file is added to the engine for the translation into multiple languages
- The lang folder is added with the dictionaries for future translations, at the moment it supports English and Spanish
- Complete translation into Spanish in the components
- In the customization menu options to modify the language are added

### Fixed
- Some fixes in the style sheets

devem-software committed:
Add Connector

### Added
- Connector vermangasporno

Sheepux commented on src/web/mjs/engine/I18n.mjs:
```diff
@@ -0,0 +1,38 @@
+export default class I18n {
```
Class wasn't working: rewrote it (you could have used a node lib)
`
export default class I18n {

    constructor() {
        this.lang = localStorage.getItem("lang") ?? "en";
        this.dic = undefined;
        this.getDictionary(`lib/hakuneko/lang/${this.lang}.json`);
    } 

    async getDictionary(url) {
        try {
            console.log('Loading translation dictionary: ',url);
            const langfile = await fetch(url);
            const tr = await langfile.json();
            this.dic = tr;
            return tr;
        } catch (e) {
            console.error('Unable to load dictionary', url);
        }
    }

    translate(key, def = '') {
        var tr = key.split('.').reduce(function(obj, key) {
            return obj && obj[key];
        }, this.dic);
        return tr ?? def;
    }
}
`

Sheepux commented on src/web/lib/hakuneko/frontend@modern/app.js:
```diff
@@ -0,0 +1,86 @@
+     *
+     */
+    getAppearance() {
+        let appearance = "";
```
This is missing a default value and results (when you have never used the modern theme) in a broken blank page. You should set a default color (indigo or blue ?)

Sheepux commented on src/web/mjs/engine/I18n.mjs:
```diff
@@ -0,0 +1,38 @@
+        this.path = require("path");
+        this.lang =
+            localStorage.getItem("lang") === null
+                ? "es"
```
"es" shouldn't be the default language but "en"

Sheepux commented on src/web/mjs/engine/I18n.mjs:
```diff
@@ -0,0 +1,38 @@
+        this.fs = require("fs");
+        this.path = require("path");
+        this.lang =
+            localStorage.getItem("lang") === null
```
can simplify default value with ??
`this.lang = localStorage.getItem("lang") ?? "en";`

Sheepux commented on src/web/lib/hakuneko/frontend@modern/components/menuComponent/script.js:
```diff
@@ -0,0 +1,181 @@
+
+    setTheme(e) {
+        localStorage.setItem("themeColor", e.model.item.value);
+        this.windowReload();
```
Would be nice to find a way to change the color theme without reloading (did it in haruneko).
Switching the base css load did the trick with svelte
`<link rel="stylesheet" href="css/{currentTheme}.css">`

devem-software committed:
Update I18n.mjs

27136773 commented on src/web/lib/hakuneko/frontend@modern/about.md:
```diff
@@ -0,0 +1,67 @@
+# Modern Theme
```
What is the purpose of this file?
I recommend to track TODOs as GitHub issues and changes are already traceable via the commit history in git.

27136773 commented on src/web/mjs/connectors/VerMangasPorno.mjs:
```diff
@@ -0,0 +1,69 @@
+import Connector from "../engine/Connector.mjs";
```
This file is not related to a change in the frontend, i recommend to make a separate PR (furthermore i think this was already added)

27136773 commented on src/web/mjs/HakuNeko.mjs:
```diff
@@ -79,6 +81,10 @@ export default class HakuNeko {
         return this._enums;
     }
 
+    get I18n() {
```
A mechanic for localization on a global level should be done in a separate PR and from my point of view the current version of HakuNeko is very unlikely to implement localization at all.

If the new frontend supports localization, i recommend a scoped solution only for the new frontend.

27136773 commented on src/web/lib/hakuneko/frontend@modern/app.css:
```diff
@@ -0,0 +1,104 @@
+  height: 100%;
+  width: 100%;
+  overflow: hidden;
+  background: var(--base);
```
Variable name is not expressive, someone who is going to change this in the corresponding file probably have no idea where this is used.
Variables should better describe their usage, e.g. `--main-panel-background-color`

27136773 commented on src/web/lib/hakuneko/frontend@modern/app.css:
```diff
@@ -0,0 +1,104 @@
+  margin-bottom: 1rem;
+  overflow: hidden;
+  border-radius: var(--border-radius);
+  border: 1px solid var(--accent);
```
Variable name is not expressive.

27136773 commented on src/web/lib/hakuneko/frontend@modern/components/bookmarksComponent/script.js:
```diff
@@ -0,0 +1,119 @@
+/* eslint-disable no-undef */
+
+class HakunekoBookmarks extends Polymer.Element {
+    /**
```
Remove empty comments (also all other places in file)

27136773 commented on src/web/lib/hakuneko/frontend@modern/components/menuComponent/script.js:
```diff
@@ -0,0 +1,181 @@
+    }
+
+    getLanguageSelected(value) {
+        return localStorage.getItem("lang") === value ? "selected" : "";
```
Key name is to common for global scope, use something like `frontend.modern.language` instead of `lang`
Also be aware that local storage is not persistent, it can be wiped randomly by the chrome engine

27136773 commented on src/web/lib/hakuneko/frontend@modern/components/menuComponent/script.js:
```diff
@@ -0,0 +1,181 @@
+    }
+
+    getCornerSelected(value) {
+        return localStorage.getItem("corner") === value ? "selected" : "";
```
Key name is to common for global scope, use something like `frontend.modern.corner` instead of `corner`
Also be aware that local storage is not persistent, it can be wiped randomly by the chrome engine

27136773 commented on src/web/lib/hakuneko/frontend@modern/components/menuComponent/script.js:
```diff
@@ -0,0 +1,181 @@
+    }
+
+    getThemeSelected(theme) {
+        return localStorage.getItem("themeColor") === theme ? "selected" : "";
```
Key name is to common for global scope, use something like `frontend.modern.themeColor` instead of `themeColor`
Also be aware that local storage is not persistent, it can be wiped randomly by the chrome engine

27136773 commented on src/web/lib/hakuneko/frontend@modern/components/menuComponent/script.js:
```diff
@@ -0,0 +1,181 @@
+    }
+
+    windowReload() {
+        window.location.reload();
```
Use `if(confirm('Change requires reload, reload now?')) { ... }` to show a dialog asking the user before reload, they may not want to do it now.

27136773 commented on src/web/lib/hakuneko/lang/en.json:
```diff
@@ -0,0 +1,181 @@
+{
```
If the localization really is supposed to be used globally (for the HakuNeko engine and all frontends), then all keys for the modern frontend should be placed inside
```json
{
  "frontend" : {
    "modern": {
      // keys go here ...
    }
  }
}
```

27136773 commented on src/web/lib/hakuneko/lang/es.json:
```diff
@@ -0,0 +1,181 @@
+{
```
If the localization really is supposed to be used globally (for the HakuNeko engine and all frontends), then all keys for the modern frontend should be placed inside
```json
{
  "frontend" : {
    "modern": {
      // keys go here ...
    }
  }
}
```

27136773 reviewed (changes requested):
My major concern would be the localization. I think it should only be done in a local scope for the modern frontend, but not be added globally to the HakuNeko engine.

May be further discussed in discord for clarification...

Also please resolve all comments which are completed or rejected (with a proper explanation why a change is not necessary), so it can be verified by the commentator.
